src/content/en/updates/2018/06/audio-worklet-design-pattern.md

```diff
@@ -0,0 +1,398 @@
+[code examples and live demos](https://googlechromelabs.github.io/web-audio-samples/audio-worklet#design-pattern)
+as well!
+
+## Recap: AudioWorklet
+
+Before diving in, let us quickly recap terms and facts around the AudioWorklet
+system which is previously introduced in
+[this post](/web/updates/2017/12/audio-worklet).
+
+  - [BaseAudioContext](https://developer.mozilla.org/en-US/docs/Web/API/BaseAudioContext):
+    Web Audio API's primary object.
+  - [AudioWorklet](https://webaudio.github.io/web-audio-api/#audioworklet) (AW):
+    A special script file loader for the AudioWorklet operation. Belongs to
+    BAC. A BAC can have one AW. The loaded script file is evaluated in the
+    AudioWorkletGlobalScope and is used to create the AudioWorkletProcessor
+    instances.
+  - [AudioWorkletGlobalScope](https://webaudio.github.io/web-audio-api/#audioworkletglobalscope)
+    (AWGS): A special JS global scope for the AudioWorklet operation. Runs on a
+    dedicated rendering thread for the WebAudio. A BAC can have one AWGS.
+  - [AudioWorkletNode](https://webaudio.github.io/web-audio-api/#audioworkletnode)
+    (AWN): An AudioNode designed for the AudioWorklet operation. Instantiated
+    from a BAC. A BAC can have multiple AWN similarly to the native AudioNodes.
+  - [AudioWorkletProcessor](https://webaudio.github.io/web-audio-api/#audioworkletprocessor)
+    (AWP): A counterpart of the `AudioWorkletNode`. The actual guts of the
+    AudioWorkletNode processing the audio stream by the user-supplied code. It
+    is instantiated in the AWGS when a AWN is constructed. An AWN can have one
+    matching AWP.
+
+## Design Patterns
+
+### Using AudioWorklet with WebAssembly
+
+  - [Example code](https://googlechromelabs.github.io/web-audio-samples/audio-worklet/design-pattern/awn/)
+
+[WebAssembly](https://developer.mozilla.org/en-US/docs/WebAssembly) is a perfect
+companion for AudioWorkletProcessor. The combination of these two features
+brings a variety of advantages to the audio processing on the web, but the two
+biggest benefits are: a) bringing existing C/C++ audio processing code into the
+WebAudio ecosystem and b) avoiding the overhead of JS JIT compilation and
+garbage collection in the audio processing code.
+
+The former is important to developers with an existing investment in audio
+processing code and libraries, but the latter is critical for nearly all users
+of the API. In the world of WebAudio, the timing budget for the stable audio
+stream is quite demanding: it is only 3ms at the sample rate of 44.1Khz. Even a
+slight hiccup in the audio processing code can cause glitches. The developer
+must optimize the code for faster processing, but also minimize the amount of JS
+garbage being generated. Using WebAssembly can be a solution that addresses both
+problems at the same time: it is faster and generates no garbage from the code.
+
+The next section describes how WebAssembly can be used with the AudioWorklet and
```

Comment:
with ~~the~~ AudioWorklet

Reply:
DBC: "used with AudioWorklet" is awkward.  "used with an AudioWorklet" or "used with AudioWorklets" reads better.